projects/npm-tools/packages/npm-scripts/src/utils/writeLibInitEntryPoint.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * SPDX-FileCopyrightText: © 2019 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const fs = require('fs');
+
+/**
+ * Create a webpack entry point containing the initialization logic for
+ * federation libraries.
+ *
+ * @param {string} filePath the path to the output file
+ *
+ */
+module.exports = function (filePath) {
+	const code = `
+const { transformChunkScriptFilename } = 
+	window[Symbol.for('__LIFERAY_WEBPACK_HELPERS__')];
+
+const __saved_webpack_get_script_filename__ = __webpack_get_script_filename__;
+
+__webpack_get_script_filename__ = (chunkId) => {
+	return transformChunkScriptFilename(
```

Comment:
So, I'm assuming we'll export `transformChunkScriptFilename` from within `DXP` to make it get in here? 

If that's the case, I don't mind the level of indirection, but I'm a bit worried about race conditions since we'll need to make sure wherever those helpers come from are already there before we run any webpack code... is that guaranteed and already in place?

Reply:
Yes it is, because if that hadn't been loaded, you could not be executing webpack code :-). All webpack code is bootstrapped in that same place.

Reply:
It will be [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-module-launcher/src/main/resources/META-INF/resources/webpack_federation.js), our "webpack loader".